_posts/en/newsletters/2024-03-20-newsletter.md

```diff
@@ -0,0 +1,160 @@
+  the [v2 protocol][topic v2 p2p transport] defined in [BIP324][].  This
+  is especially intended to allow light client wallets written for v1 to
+  take advantage of v2's traffic encryption.
+
+  Light clients typically only announce transactions belonging to their
+  own wallets, so anyone capable of eavesdropping on an unencrypted v1
+  connection can reasonably conclude that a transaction sent by a light
+  client belonged to someone using the origin IP address.  When v2
+  encryption is used, only the full nodes receiving the transaction will
+  be able to definitively identify it as originating from the light
+  client's IP address, assuming none of the light client connections is
+  subject to a man-in-the-middle attack (which is possible to detect in
+  some cases and which [later upgrades][topic countersign] may
+  automatically defend against).
+
+  Falbesoner's initial work pulls together BIP324 functions written in
+  Python for Bitcoin Core's testing suite, which results in a proxy that
+  is "terribly slow and vulnerable to side-channel attacks [and] not
+  recommended to use it for anything but tests right now".  However, he
+  is working on rewriting the proxy in Rust and may also make some or
+  all of its functions available as a library for light clients or other
+  software that wants to natively support the v2 Bitcoin P2P protocol.
+
+- **Overview of BTC Lisp:** Anthony Towns [posted][towns lisp] to
+  Delving Bitcoin about his experiments over the past couple years
+  creating a variant of the [Lisp][] language for Bitcoin, called BTC
+  Lisp.  See Newsletters [#293][news293 lisp] and [#191][news191 lisp]
+  for previous discussions.  The post goes into significant detail; we
+  encourage anyone interested in the idea to read it directly.  We will
+  briefly quote from its _conclusion_ and _future work_ sections:
+
+  "[BTC Lisp] can be a little expensive on-chain, but it seems like you
+  can do pretty much anything [...] I don’t think implementing either a
+  Lisp interpreter or the bucket of opcodes that would need to accompany
+  it is too hard [but] it is pretty annoying to write Lisp code without
+  a compiler translating from a higher level representation down to the
+  consensus-level opcodes, [though] that seems solvable.  [T]this could
+  be taken further [by] implementing a language like this and deploying
+  it on signet/inquisition."
+
+  Russell O'Connor, developer of the [Simplicity][topic simplicity]
+  language that may also one day be considered as an alternative
+  consensus scripting language, [replied][oconnor lisp] with some
+  comparisons between Bitcoin's current Script language, Simplicity, and
+  Chia/BTC Lisp.  He concludes, "Simplicity and the [Chia Lisp Virtual
+  Machine] clvm are both low level languages that are meant to be easy
+  for machines to evaluate, which causes tradeoffs that make them hard
+  for humans to read. They are intended to be the compiled from some
```

Comment:
```suggestion
  for humans to read. They are intended to be compiled from some
```
This typo is within a quotation, so I'm unsure what to do here.

Reply:
Fixed it.  I don't like changing people's unique language, but I'm totally ok with fixing typos and making other changes that are 100% clearly following their intent, in non-adversarial situations.  If this was an adversarial case, we wouldn't make any change.